### DISCUSSION TITLE
Hide from CDN eyes?!

### DISCUSSION THREAD

**Discussion by APT-ZERO** [XTLS/Xray-core#3648] [Q&A] [answered]
Some Iranian people use Iranian CDNs for proxy
as those CDNs traffic are not free and they make a lot of money from this proxy users-
they don't care about usage of their CDN for proxy
But if they decide to block proxy traffic, is it possible for users to stay hidden from CDN eyes too?

**Comment by mmmray** [comment, marked as answer]
no core that I know of *really* considers this as part of their threat model, and no transport in xray is designed with this requirement in mind. it's difficult to achieve for the same reasons notls cannot be made undetectable in the same way that TLS can (see https://github.com/XTLS/Xray-core/discussions/3582#discussioncomment-10119641)

the best you can do today is to hide the user traffic with vmess, but that's just for personal security, not for proxy detection.

if it is achieved anywhere, then I think it will come at a severe performance cost.
